falcon/util/structures.py

```diff
@@ -108,22 +108,114 @@ def __repr__(self):
 
 
 class ETag(str):
-    """An entity-tag ``str``-like object with the weakness indicator.
+    """Convenience class to represent a parsed HTTP entity-tag.
+
+    This class is simply a subclass of ``str`` with a few helper methods and
+    an extra attribute to indicate whether the entity-tag is weak or strong. The
+    value of the string is equivalent to what RFC 7232 calls an "opaque-tag",
+    i.e. an entity-tag sans quotes and the weakness indicator.
+
+    Note:
+
+        Given that a weak entity-tag comparison can be performed by
+        using the ``==`` operator (per the example below), only a
+        :meth:`~.strong_compare` method is provided.
+
+    Here is an example ``on_get()`` method that demonstrates how to use instances
+    of this class::
+
+        def on_get(self, req, resp):
+            content_etag = self._get_content_etag()
+            for etag in (req.if_none_match or []):
+                if etag == '*' or etag == content_etag:
+                    resp.status = falcon.HTTP_304
+                    return
+
+            # ...
+
+            resp.etag = content_etag
+            resp.status = falcon.HTTP_200
+
+    (See also: RFC 7232)
 
     Attributes:
-        is_weak (bool): The weakness indicator.
+        is_weak (bool): ``True`` if the entity-tag is weak, otherwise ``False``.
 
     """
+
     is_weak = False
 
-    def to_header(self):
-        """Convert the ETag into a HTTP header string.
+    def strong_compare(self, other):
+        """Performs a strong entity-tag comparison.
+
+        Two entity-tags are equivalent if both are not weak and their
+        opaque-tags match character-by-character.
+
+        (See also: RFC 7232, Section 2.3.2)
+
+        Arguments:
+            other (ETag): The other :class:`~.ETag` to which you are comparing
+            this one.
+
+        Returns:
+            bool: ``True`` if the two entity-tags match, otherwise ``False``.
+
+        """
+
+        return self == other and not (self.is_weak or other.is_weak)
+
+    def dumps(self):
```

Comment:
I wasn't sure what to name these methods, so I just defaulted to something that hopefully seems pythonic. Suggestions welcome if these names don't make sense.